### PR TITLE
Addtocartbutton  functiooning as disired Redux Persist

### PR DESCRIPTION
**Pull request by yelab07**
"addToCart button functioning as desired and redux prsist implimented "

## Changes
_List Changes Introduced by this PR_
1. AddToCart  button working as desired reflecting the item with the quantity in our store state
2. Redux Persist implemented so we have improved user experience. 

## Purpose
We wanted  for the quantity button in our productdtailpage to work seamlessly with the addtocart button reflecting the selected quantity mount to be stored in our state for each item increasing the number in store whenever the same item is added in addition to what is already added.
We also needed to persist our state so that if users close the app and come back they will still see the previous value which is very important to enhance the user experience 

## Learning
It was first time experience with Redux persist which works along side  with react native AsyncStotage so we can leverage local storage [Redux Persist](https://javascript.plainenglish.io/integrating-redux-toolkit-redux-persist-and-react-native-debugger-to-a-react-native-app-f41e63476016)

### CODE REVIEW

File: screens/Home/ProductDetailsScreen.tsx [hunk 1]
```diff
@@ -8,25 +8,34 @@ import {
...
   const navigation = useNavigation();
   const route = useRoute<RouteProp<ParamList, "Params">>();
   const { id } = route.params;
   const { data } = useProductQuery(id);
   const [quantity, setQuantity] = useState<any>(1);
 
-  const onAddToCart = () => { };
+  const onAddToCart = (product: Product, quantity: number) => {
+    dispatch(addToCart({ ...product, quantity }))
+    console.log(quantity)
```
Comment (kmausolf) [inline screens/Home/ProductDetailsScreen.tsx]: Console log here

File: store.tsx [hunk 2]
```diff
@@ -1,12 +1,23 @@
...
+  version: 1,
+  storage: AsyncStorage,
+}
+const rootReducer = combineReducers({
   cart: cartReducer,
+})
+const persistedReducer = persistReducer(persistConfig, rootReducer)
+
+const allReducers = combineReducers({
+  [productsApi.reducerPath]: productsApi.reducer, persistedReducer,
```
Comment (kmausolf) [inline store.tsx]: Nice job getting this figured out!  So, the guide you used is likely out of date considering that you're using Redux Toolkit.  Here's a Redux Toolkit version of the above that should work and be easier to interact with:

```
const persistedCartReducer = persistReducer(persistConfig, cartReducer);

export const store = configureStore({
  reducer: {
    cart: persistedCartReducer,
    [productsApi.reducerPath]: productsApi.reducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(productsApi.middleware),
});
```

You don't need to combine the reducers now.  That's something we used to have to do when Redux only had a single reducer entry point.  Hope this helps reduce the complexity of your app!

**Review by kmausolf — COMMENTED** [review]
Great work on this, getting `redux-persist` to work with Redux Toolkit can be a challenge, but I've got a suggestion that could simplify things quite a bit.

File: store.tsx (hunk 2 above)
Comment (yelab07) [reply store.tsx]: Thanks so much Kendrick for your feedback I will make the changes accordingly !

File: store.tsx [hunk 3]
```diff
@@ -1,12 +1,23 @@
 import { configureStore, getDefaultMiddleware } from "@reduxjs/toolkit";
 import { productsApi } from "./services/productsApi";
 import { cartReducer } from "./redux/cartSlice";
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import { persistStore, persistReducer } from 'redux-persist'
```
Comment (eangele1) [inline store.tsx]: Interesting use of Redux Persist! Please make sure that the cart state continues to function on mobile as well as web if desired. Other than that, good work!

File: store.tsx [hunk 4]
```diff
@@ -14,16 +21,21 @@ const persistConfig = {
...
+  reducer: {
+    cart: persistedCartReducer,
+    [productsApi.reducerPath]: productsApi.reducer,
+  },
   middleware: (getDefaultMiddleware) =>
-    getDefaultMiddleware().concat(productsApi.middleware),
+    getDefaultMiddleware({
+      serializableCheck: {
+        ignoredActions: [FLUSH, REHYDRATE, PAUSE, PERSIST, PURGE, REGISTER],
+      },
```
Comment (kmausolf) [inline store.tsx:37]: Good find!  Sorry that I had forgotten to mention this!

**Review by kmausolf — APPROVED** [review]

File: store.tsx (hunk 3 above)
Comment (yelab07) [reply store.tsx]: Thank you Ezer!

**Pull request merged**